internal/app/api/server/private/agents.go

```diff
@@ -265,14 +258,22 @@ func (s *AgentService) GetAgents(c *gin.Context) {
 		usePolicy   bool
 	)
 
-	if err = c.ShouldBindQuery(&query); err != nil {
+	if err := c.ShouldBindQuery(&query); err != nil {
 		utils.FromContext(c).WithError(err).Errorf("error binding query")
 		utils.HTTPError(c, srverrors.ErrGetAgentsInvalidRequest, err)
 		return
 	}
 
-	if iDB = utils.GetGormDB(c, "iDB"); iDB == nil {
-		utils.HTTPError(c, srverrors.ErrInternalDBNotFound, nil)
+	serviceHash, ok := srvcontext.GetString(c, "svc")
+	if !ok {
+		utils.FromContext(c).Errorf("could not get service hash")
+		utils.HTTPError(c, srverrors.ErrInternal, nil)
+		return
+	}
+	iDB, err := s.serverConnector.GetDB(c, serviceHash)
```

Comment:
It looks like this pattern is common for majority of cases, maybe it is better to add additional functions to serverConnector, like:
```
serverConnector.GetDBForContext(c)
```
wdyt?

Reply:
I think that we should not couple connector and handlers by gin context, because the way how you fetch agent server hash can be different, and connector can be used in places like workers too. A little verbosity here by now is ok, there is always room for improvements, my next PR will target user action logging, so let's review that after final PR for handlers refactoring.